authz-module/index.md

```diff
@@ -0,0 +1,204 @@
+
+## Requirements
+
+Before you start the tutorial, you need to install the `simd` binary.
+
+Clone the `cosmos-sdk` repository:
+```
+git clone https://github.com/cosmos/cosmos-sdk
+```
+
+Change directories and check out `v0.43.0`:
+```
+cd cosmos-sdk && git checkout v0.43.0
+```
+
+Install the `simd` binary:
+```
+make install
+```
+
+Check to make sure the installation was successful:
+```
+simd version
+```
+
+You should see `0.43.0` printed to the console.
+
+## Configuration
+
+<!-- TODO: update tip to use `unsafe-reset-all` with better user experience -->
+
+::: tip
+If you have used `simd` before, you might already have a `.simapp` directory in your home directory. To keep the previous data, either save the directory to another location or use the `--home` flag and specify a different directory for each command in the following instructions. If you don't want to keep the previous data, remove the previous directory (`rm -rf ~/.simapp`).
+:::
+
+Run the following commands to configure the `simd` binary.
+
+Set the chain ID:
+```
+simd config chain-id demo
+```
+
+Set the keyring backend:
+```
+simd config keyring-backend test
+```
+
+## Key Setup
+
+Create a couple of test keys. Use the `--recover` option so that the addresses used in the following example commands are consistent.
+
+Add a key for Alice:
+```
+simd keys add alice --recover
+```
+
+Enter the following mnemonic:
+```
+plunge hundred health electric victory foil marine elite shiver tonight away verify vacuum giant pencil ocean nest pledge okay endless try spirit special start
+```
+
+Add a key for Bob:
+```
+simd keys add bob --recover
+```
+
+Enter the following mnemonic:
+```
+shuffle oppose diagram wire rubber apart blame entire thought firm carry swim old head police panther lyrics road must silly sting dirt hard organ
+```
+
+## Chain Setup
+
+The following commands start a chain for the simulation application (`simapp`).
+
+Initialize the node:
+```
+simd init test --chain-id demo
+```
+
+Add Alice and an initial balance to the genesis file:
+```
+simd add-genesis-account alice 5000000000stake
+```
+
+Add Bob and an initial balance to the genesis file:
+```
+simd add-genesis-account bob 5000000000stake
+```
+
+Generate a transaction adding Alice to the initial validator set:
+```
+simd gentx alice 1000000stake --chain-id demo
+```
+
+Add the validator transaction to the genesis file:
+```
+simd collect-gentxs
+```
+
+## Start Chain
+
+You are now ready to start a single node network on your local machine.
+
+Start the chain:
+```
+simd start
+```
+
+## Submit Proposal
+
+To demonstrate an authorization to vote on a governance proposal, you must first create a governance proposal.
+
+Create proposal:
+```
+simd tx gov submit-proposal --title="Test Authorization" --description="Is Bob authorized to vote?" --type="Text" --deposit="10000000stake" --from alice
```

Comment:
It's not clear to me exactly why a deposit is needed here. Maybe we should clarify? Or is this something obvious that I'm missing?

Reply:
Each proposal requires a deposit and `10000000stake` is the minimum deposit. Including it here prevents the need to make the deposit in a separate transaction. Once the minimum is reached, the proposal enters the voting period.

Reply:
Ah yeah that makes total sense, thanks. Was just wondering if that's something we want to mention, as in, elaborating a little bit more on the parameters of this (and maybe other) command(s)?

Reply:
as a learner for go commands (that are positional and do not require parameters before values) is to use the where method (although here it looks like these -- are flags, not parameters?) I always want to know more about flags, and the help doesn't show them (or I don't know how to make the help show all the flags available)

Reply:
> Was just wondering if that's something we want to mention, as in, elaborating a little bit more on the parameters of this (and maybe other) command(s)?

Sure, I can do that. This command is not directly related to the authz module but it wouldn't hurt to provide more information about it in this tutorial.

> I always want to know more about flags, and the help doesn't show them (or I don't know how to make the help show all the flags available)

You can view all flags available using `--help` (i.e. `simd tx gov submit-proposal --help`).